echopop/utils/load.py

```diff
@@ -414,17 +427,34 @@ def validate_data_columns(
             # column names of the workbook
             if "vario_krig_para" in config_map:
                 data_columns = [list(row) for row in zip(*sheet.iter_rows(values_only=True))][0]
+                # ---- Define an empty list
+                valid_columns = []
             else:
-                data_columns = {col.value for col in sheet[1]}
-
-            # Error evaluation and print message (if applicable)
-            if not set(validation_settings.keys()).issubset(set(data_columns)):
-                missing_columns = set(validation_settings.keys()) - set(data_columns)
-                raise ValueError(f"Missing columns in the Excel file: {missing_columns}")
+                # ---- Get actual column names
+                original_columns = [col.value for col in sheet[1]]
+                # ---- Translate the column names, if needed
+                data_columns = [
+                    NAME_CONFIG[name] if name in NAME_CONFIG else name for name in original_columns
+                ]
+
+                # Error evaluation and print message (if applicable)
+                if not set(validation_settings.keys()).issubset(set(data_columns)):
+                    missing_columns = set(validation_settings.keys()) - set(data_columns)
+                    raise ValueError(f"Missing columns in the Excel file: {missing_columns}")
+
+                # Get present data column names
+                valid_columns = [
+                    o
+                    for o, t in zip(original_columns, data_columns)
+                    if t in validation_settings.keys()
+                ]
```

Comment:
This section gets the job done, but it is a little convoluted because you need to get the new column names but then have to fish out the corresponding old column names in order to pass those to `read_validated_data`.

Wouldn't it be more straightforward to combine this function and `read_validated_data` so that you only read the sheets once and can convert the column names directly as they are read in, and check if anything is missing there?